service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1907,13 +1907,15 @@ public TariffsForLocationDto getTariffForOrder(Long id) {
     public Set<String> getAllAuthorities(String email) {
         Employee employee = employeeRepository.findByEmail(email)
             .orElseThrow(() -> new NotFoundException(EMPLOYEE_DOESNT_EXIST));
-        return userRemoteClient.getAllAuthorities(employee.getEmail());
+        Set<String> authorities = userRemoteClient.getAllAuthorities(employee.getEmail());
+        if (authorities.isEmpty()) {
+            throw new NotFoundException(COULD_NOT_RETRIEVE_EMPLOYEE_AUTHORITY);
+        }
+        return authorities;
     }
 
     @Override
     public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String email) {
-        Employee employee = employeeRepository.findByEmail(dto.getEmployeeEmail())
-            .orElseThrow(() -> new NotFoundException(EMPLOYEE_DOESNT_EXIST));
```

Comment:
Why did you remove this?
Is it possible to update authorities for a non-existent employee?